### PR TITLE
Add opponent organisations feature flag

### PR DESCRIPTION
**Pull request by skdejong**
## What

[Add organisation feature flag to Apply](https://dsdmoj.atlassian.net/browse/AP-4278)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: nly [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Inspecting 1 file
+.
+
+0 files inspected, no offenses detected
```
Comment (naseberry) [inline nly:4]: Should this file be ignored?

Comment (jsugarman) [reply nly:4]: looks like output from rubocop or other linter/tool. will need to be removed I think

File: spec/models/setting_spec.rb [hunk 2]
```diff
@@ -54,6 +54,7 @@
       expect(described_class.alert_via_sentry?).to be true
```
Comment (kmahern) [inline spec/models/setting_spec.rb:54]: @skdejong I think you could add a test for opponent_organisations? to the "when there is no setting record" and "when a record already exists with non-default values" contexts in this spec - give me a shout if you want to pair 🙂

File: spec/models/setting_spec.rb [hunk 3]
```diff
@@ -54,6 +54,7 @@
...
       expect(described_class.means_test_review_phase_one?).to be false
       expect(described_class.partner_means_assessment?).to be false
+      expect(described_class.opponent_organisations?).to be false
```
Comment (jsugarman) [inline spec/models/setting_spec.rb:57]: do we also need tests above - for example, wherever `partner_means_assessment` is tested we should have equivalent test for this flag I think

Comment (jsugarman) [reply spec/models/setting_spec.rb:57]: I would follow along with changes from this PR to make sure everything is covered in terms of specs 😄 
https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/5044/files

**Review by jsugarman — COMMENTED** [review]
See comments regarding specs plus do we need to amend `lib/tasks/setup_db_for_testing.rake`? I am not sure whether we even need that task anymore though

File: nly (hunk 1 above)
Comment (skdejong) [reply nly]: I have deleted this file now. Thanks for pointing it out.

File: spec/models/setting_spec.rb (hunk 2 above)
Comment (skdejong) [reply spec/models/setting_spec.rb:57]: Thank you, Katharine. I have added these tests now.

File: spec/models/setting_spec.rb (hunk 3 above)
Comment (skdejong) [reply spec/models/setting_spec.rb:60]: Thank you, Joel. I have added these tests now.

**Review by jsugarman — APPROVED** [review]
LGTM 👍

**Pull request merged**